packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -208,36 +208,24 @@ class FlutterDevice {
   }
 
   Future<void> refreshViews() async {
-    if (vmServices == null || vmServices.isEmpty) {
-      return Future<void>.value(null);
+    if (vmService == null) {
+      return;
     }
-    final List<Future<void>> futures = <Future<void>>[
-      for (VMService service in vmServices) service.vm.refreshViews(waitForViews: true),
-    ];
-    await Future.wait(futures);
+    return vmService.vm.refreshViews(waitForViews: true);
   }
 
   List<FlutterView> get views {
-    if (vmServices == null) {
+    if (vmService == null || vmService.isClosed) {
       return <FlutterView>[];
     }
 
-    return vmServices
-      .where((VMService service) => !service.isClosed)
-      .expand<FlutterView>(
-        (VMService service) {
-          return viewFilter != null
-               ? service.vm.allViewsWithName(viewFilter)
-               : service.vm.views;
-        },
-      )
-      .toList();
+    return (viewFilter != null
+        ? vmService.vm.allViewsWithName(viewFilter)
+        : vmService.vm.views).toList();
   }
 
   Future<void> getVMs() async {
-    for (VMService service in vmServices) {
-      await service.getVM();
-    }
+    return vmService.getVM();
```

Comment:
does this not have to be `await`ed?

Reply:
returning a future is approximately the same as awaiting a future and then returning

Reply:
Maybe:
```dart
Future<void> getVMs() => vmService.getVM();
```

Reply:
Done